### ISSUE TITLE
`object-shorthand` fails on calculated property

### ISSUE THREAD

**Issue by nkbt** [eslint/eslint#2937]
```js
commands = OrderedMap({[id]: id}).concat(commands);
```

```sh
src/stores/CommandStore.js
  78:27  error  Expected property shorthand  object-shorthand
```

Parser is default, not changed.

**Comment by xjamundx** [comment]
I have verified this bug and agree that this is a problem.

Here's what I'm seeing on latest master with this file called `test2.js`:

```js
var y = "26";
var x = {[y]: y};
```

Run like this: `node ./bin/eslint.js --rule '{object-shorthand: 2}' --env es6 test2.js `

```
  1:1   error  Use the global form of "use strict"  strict
  2:5   error  x is defined but never used          no-unused-vars
  2:10  error  Expected property shorthand          object-shorthand
```

**Comment by xjamundx** [comment]
Technically this could be expected to be shorthand `var x = {['y']: y}`, but I feel like it's too odd to worry about, so I'm just going to exclude all compute properties.
